### PR TITLE
Add dash option to curvature

### PR DESCRIPTION
**Pull request by v0d1ch**

### CODE REVIEW

File: cairo/src/Slay/Cairo.hs [hunk 1]
```diff
@@ -116,6 +116,7 @@ instance RenderElement g (PrimCurve g) where
           (x + fst pEnd)   (y + snd pEnd)
     Cairo.moveTo (x + x1) (y + y1)
     curveThrough p1 p2 p3
+    Cairo.setDash (map realToFrac dashList) (realToFrac dashStrokeOffset)
```
Comment (int-index) [inline cairo/src/Slay/Cairo.hs]: `realToFrac` shouldn't be used now that I have abstracted coordinate types

File: cairo/src/Slay/Cairo/Prim.hs [hunk 2]
```diff
@@ -84,7 +84,9 @@ data PrimCurve g =
     { curveExtents :: Extents,
       curveCurvature :: g Curvature,
       curveColor :: g Color,
-      curveDirection :: g Direction
+      curveDirection :: g Direction,
+      curveDash :: [Natural],
```
Comment (int-index) [inline cairo/src/Slay/Cairo/Prim.hs]: use `Unsigned` instead of `Natural`, add support for animations

File: cairo/src/Slay/Cairo/Prim.hs [hunk 3]
```diff
@@ -84,7 +84,9 @@ data PrimCurve g =
...
+      curveDashOffset :: Natural
```
Comment (int-index) [inline cairo/src/Slay/Cairo/Prim.hs]: `g Unsigned`

Comment (int-index) [reply cairo/src/Slay/Cairo/Prim.hs]: Actually, should the offset be non-negative? If it's not a requirement, consider making it `Signed`

File: gtk/src/Slay/Gtk.hs [hunk 4]
```diff
@@ -369,7 +369,7 @@ exampleLayout = mkLayout $ Vis $
       substrate (LRTB 5 5 5 5) (rect $ PhaseColor $ \colorPhase -> rgb colorPhase 130 200) $
       substrate (LRTB 1 1 1 1) (rect $ PhaseConst $ rgb 0 0 0) $
       substrate (LRTB 3 3 3 3) (rect $ PhaseConst $ rgb 255 255 255) $
-      substrate (LRTB 3 3 3 3) (curve (PhaseCurvature Curvature) (PhaseColor $ \colorPhase -> rgb colorPhase 130 200) (PhaseConst (Direction True False)) ) $
+      substrate (LRTB 3 3 3 3) (curve (PhaseCurvature Curvature) (PhaseColor $ \colorPhase -> rgb colorPhase 130 200) (PhaseConst (Direction True False)) ([(2 :: Natural), (3 :: Natural), (3 :: Natural) , (5 :: Natural)]) (2 :: Natural) ) $
```
Comment (int-index) [inline gtk/src/Slay/Gtk.hs]: Did you make sure an infinite list could be used here?

Comment (v0d1ch) [reply gtk/src/Slay/Gtk.hs]: They don't for now
```
src/Slay/Gtk.hs:372:168: error:
    • Could not deduce (Enum Unsigned)
        arising from the arithmetic sequence ‘2 .. ’
      from the context: HasView s El a
        bound by a type expected by the context:
                   forall s a. HasView s El a => Vis (Collage s a)
        at src/Slay/Gtk.hs:(362,17)-(380,36)
      or from: reflection-2.1.2:Data.Reflection.Reifies s1 (View El a1)
        bound by the inferred type of
                 mkMsgbox :: reflection-2.1.2:Data.Reflection.Reifies
                               s1 (View El a1) =>
                             Text -> Collage s1 a1
        at src/Slay/Gtk.hs:(368,5)-(376,72)
    • In the first argument of ‘PhaseConst’, namely
        ‘([2 .. ] :: [Unsigned])’
      In the fourth argument of ‘curve’, namely
        ‘(PhaseConst ([2 .. ] :: [Unsigned]))’
      In the second argument of ‘substrate’, namely
        ‘(curve
            (PhaseCurvature Curvature)
            (PhaseColor $ \ colorPhase -> rgb colorPhase 130 200)
            (PhaseConst (Direction True False))
            (PhaseConst ([2 .. ] :: [Unsigned]))
            (2 :: Signed))’
    
```

**Pull request closed without merging**